### ISSUE TITLE
V4 | The buyer cannot select the quantity purchased from the order

### ISSUE THREAD

**Issue by boulinb** [trader-xyz/nft-swap-sdk#95]
Hello everyone!

I noticed that the user, when he bought an ERC1155 against an ERC20 could not choose the quantity bought; If the offer is 5 ETH against 5 ERC1155, it is impossible from the SDK to buy 1 ERC1155 for 1 ETH.

I have fork your project and create a fix, not being allowed to make a PR, here is a screen of the patch : 

(src/sdk/v4/NftSwapV4.ts): https://prnt.sc/c1G4i5T-Z13m
(src/sdk/v4/types.ts) : https://prnt.sc/B4Fw-ZR2Zd2w

**Comment by saharshkhicha18** [comment]
Would want this to be integrated

**Comment by nedgar** [comment]
I agree. Since 0x allows a taker to take fewer than the maker offered, the NftSwap SDK should support this and take an optional quantity in the options when filling a sell order.

**Comment by jinsley8** [comment]
> I have fork your project and create a fix, not being allowed to make a PR, here is a screen of the patch :
> 
> (src/sdk/v4/NftSwapV4.ts): https://prnt.sc/c1G4i5T-Z13m (src/sdk/v4/types.ts) : https://prnt.sc/B4Fw-ZR2Zd2w

@boulinb Have you tested that your fix works? If it does then I agree this would be amazing to merge.

**Comment by boulinb** [comment]
Hey @jinsley8 ! 

Yes I use it in production on my project ;)

**Comment by jinsley8** [comment]
What happens to the original order after a partial buy?

Can you still do another order from it or do you just cancel the remaining order?

**Comment by boulinb** [comment]
In my case, the order is partially filled and waiting to be completely filled: If I put 2 ERC-1155s up for sale, I can buy 1 and then buy the rest back with another transaction
